stakingModel/test/BasicCalculation.test.js

```diff
@@ -12,144 +12,131 @@ contract("InterestDistribution - Basic calculations", ([user1]) => {
-    expect(yieldData[1].toString()).to.be.equal("2");
+      * GlobalYieldPerToken = GlobalYieldPerToken(P) + GDEarnedInterest/GlobalTotalEffectiveStake.
+      * 500 + (270)/27 = 500 + 10 => 510 x 1e11 (27 + 2(G$ precision) - 18(token decimal) = 11 precision points)
+      */ 
+    expect((Math.round(yieldData[0]/1e11)).toString()).to.be.equal("510");
   });
 
   it("Should return correct G$ interest", async () => {
 
     /**
       * Formula: 
-      * EarnedGDInterest = MAX[TotalStaked x (AccumulatedYieldPerDAI - AvgYieldRatePerDAI) - WithdrawnToDate, 0]
+      * EarnedGDInterest = MAX[TotalEfectiveStaked x AccumulatedYieldPerDAI - (GDYieldRate + WithdrawnToDate), 0]
       * consider :
-      * AvgYieldRatePerToken(P) = 2
-      * AccumulatedYieldPerToken(P) = 5
-      * TotalStaked = 36 x 1e18
-      * WithdrawnToDate = 0;
-      * output = 180 x 1e2
+      * GDYieldRate(P) = 3500 x 1e29
+      * AccumulatedYieldPerToken(P) = 510 x 1e11
+      * TotalEfectiveStaked = 27 x 1e18
+      * WithdrawnToDate = 0
+      * output = 10270 x 1e2
       */
 
       let earnedGDInterest = await interestDistribution.calculateGDInterest(user1);
       
       /**
       * Formula: 
-      * EarnedGDInterest = MAX[TotalStaked x (AccumulatedYieldPerDAI - AvgYieldRatePerDAI) - WithdrawnToDate, 0]
-      * Max[36 * (5 - 2) - 0, 0] = Max[36 * 3, 0] = Max[108, 0] = 108 = 2 points presicion 10800
+      * EarnedGDInterest = MAX[TotalEfectiveStaked x AccumulatedYieldPerDAI - (GDYieldRate + WithdrawnToDate), 0]
+      * Max[27 * 510 - (3500 + 0), 0] = Max[13770 - 3500, 0] = Max[10207, 0] = 10270 = 2 points presicion 1027000
       */
-      expect(earnedGDInterest.toString()).to.be.equal("10800");
+      expect(earnedGDInterest.toString()).to.be.equal("1027000");
   });
 
-  it("Should return G$ interest as 0 if WithdrawnToDate > TotalStaked x (AccumulatedYieldPerToken - AvgYieldRatePerToken)", async () => {
+  it("Should return G$ interest as 0 if (GDYieldRate + WithdrawnToDate) > TotalEfectiveStaked x AccumulatedYieldPerDAI", async () => {
 
 
     // withdrawing G$ interest 
     await interestDistribution
-        .updateWithdrawnInterest(user1)
+        .withdrawGDInterest(user1)
         .catch(e => e);
 
-
     let stkerData = await interestDistribution.getStakerData(user1);
-    expect(stkerData[3].toString()).to.be.equal("10800");
+    expect(stkerData[3].toString()).to.be.equal("1027000");
 
     /**
-      * Stake = 7.2 x 10^18
-      * Donation = 0%
-      * IntrestTokenRate = 1.03
-      * LastTokenRate = 1.03
-      * IntrestTokenHoldings = 60 x 1e18
+      * Effective stake
       */
-    // Staking so we can manipulate AvgYieldRatePerDAI
+    // Manipulate GDYieldRate to create scenario.
     await interestDistribution
-        .stakeCalculation(user1, web3.utils.toWei("7.2", "ether"), 0, web3.utils.toWei("1.03", "ether"), web3.utils.toWei("60", "ether"))
+        .updateGDYieldRate(user1, web3.utils.toWei("3", "ether"))
```

Comment:
GDYieldRate needs to be renamed
maybe to "stakeBuyinRate"

Reply:
renamed